x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/related_integrations/related_integrations.cy.ts

```diff
@@ -221,14 +221,29 @@ describe('Related integrations', { tags: ['@ess', '@brokenInServerless'] }, () =
         openTable();
         filterBy(RELATED_INTEGRATION_FIELD);
 
-        RULE_RELATED_INTEGRATIONS.forEach((integration) => {
-          cy.contains(
-            FIELD(RELATED_INTEGRATION_FIELD),
-            `{"package":"${integration.package}"${
-              integration.integration ? `,"integration":"${integration.integration}"` : ''
-            },"version":"${integration.version}"}`
-          );
-        });
+        cy.get(FIELD(RELATED_INTEGRATION_FIELD))
+          .invoke('text')
+          .then((stringValue) => {
+            // Integrations are displayed in the flyout as a string with a format like so:
+            // '{"package":"aws","version":"1.17.0","integration":"unknown"}{"package":"mock","version":"1.1.0"}{"package":"system","version":"1.17.0"}'
+            // We need to parse it to an array of valid objects before we can compare it to the expected value
+            // Otherwise, the test might fail because of the order of the properties in the objects in the string
+            const jsonStringArray = stringValue.split('}{');
+
+            const validJsonStringArray = createValidJsonStringArray(jsonStringArray);
+
+            const parsedIntegrations = validJsonStringArray.map((jsonString) =>
+              JSON.parse(jsonString)
+            );
```

Comment:
A follow-up to the above comment: e.g., if this `"}{"` formatting changes in the Alert Details flyout, that would cause this test to fail, and this would be a false negative outcome because the feature would still work properly.